stdlib/SuiteSparse/src/cholmod.jl

```diff
@@ -248,20 +248,20 @@ struct C_SparseVoid <: SuiteSparseStruct
 end
 
 mutable struct Sparse{Tv<:VTypes} <: AbstractSparseMatrix{Tv,SuiteSparse_long}
-    p::Ptr{C_Sparse{Tv}}
-    function Sparse{Tv}(p::Ptr{C_Sparse{Tv}}) where Tv<:VTypes
-        if p == C_NULL
+    ptr::Ptr{C_Sparse{Tv}}
+    function Sparse{Tv}(ptr::Ptr{C_Sparse{Tv}}) where Tv<:VTypes
+        if ptr == C_NULL
             throw(ArgumentError("sparse matrix construction failed for " *
                 "unknown reasons. Please submit a bug report."))
         end
-        A = new(p)
+        A = new(ptr)
         finalizer(free!, A)
         return A
     end
 end
 Sparse(p::Ptr{C_Sparse{Tv}}) where {Tv<:VTypes} = Sparse{Tv}(p)
 
-Base.unsafe_convert(::Type{Ptr{Tv}}, A::Sparse{Tv}) where {Tv} = A.p
+Base.unsafe_convert(::Type{Ptr{Tv}}, A::Sparse{Tv}) where {Tv} = getfield(A, :ptr)
```

Comment:
Switch necessary?

Reply:
I had to change the name of the field to `ptr` because `F.p` is now syntax for extracting a permutation vector.

Reply:
Nice!